platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java

```diff
@@ -37,21 +42,42 @@ public class SimultaneousSyncThrottle {
     /** lock per each other member, each one is used by all caller threads and the listener thread */
     private final Map<Long, SyncLock> simSyncThrottleLock;
 
-    public SimultaneousSyncThrottle(final int maxListenerSyncs) {
+    private static final RunningAverageMetric.Config AVG_SIM_SYNCS_CONFIG = new RunningAverageMetric.Config(
+                    PLATFORM_CATEGORY, "simSyncs")
+            .withDescription("avg number of simultaneous syncs happening at any given time")
+            .withFormat(FORMAT_9_6);
+
+    private static final RunningAverageMetric.Config AVG_SIM_LISTEN_SYNCS_CONFIG = new RunningAverageMetric.Config(
+                    PLATFORM_CATEGORY, "simListenSyncs")
+            .withDescription("avg number of simultaneous listening syncs happening at any given time")
+            .withFormat(FORMAT_9_6);
+
+    /**
+     * Construct a new SimultaneousSyncThrottle
+     *
+     * @param metrics          the metrics engine
+     * @param maxListenerSyncs the maximum number of listener syncs that can happen at the same time
+     */
+    public SimultaneousSyncThrottle(@NonNull final Metrics metrics, final int maxListenerSyncs) {
         this.maxListenerSyncs = maxListenerSyncs;
         simSyncThrottleLock = new ConcurrentHashMap<>();
+
+        final RunningAverageMetric avgSimSyncs = metrics.getOrCreate(AVG_SIM_SYNCS_CONFIG);
```

Comment:
null check

Reply:
This is dereferenced in the constructor and will throw an exception promptly if null.